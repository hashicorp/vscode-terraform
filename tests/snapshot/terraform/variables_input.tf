# input variables

variable "image_id" {
  type = string
}

variable "availability_zone_names" {
  type    = list(string)
  default = ["us-west-1a"]
}

variable "docker_ports" {
  type = list(object({
    internal = number
    external = number
    protocol = string
  }))
  default = [
    {
      internal = 8300
      external = 8300
      protocol = "tcp"
    }
  ]
}

variable "image_id" {
  type        = string
  description = "The id of the machine image (AMI) to use for the server."

  validation {
    condition     = length(var.image_id) > 4 && substr(var.image_id, 0, 4) == "ami-"
    error_message = "The image_id value must be a valid AMI id, starting with \"ami-\"."
  }
}
