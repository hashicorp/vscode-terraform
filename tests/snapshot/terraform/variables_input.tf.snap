># input variables
#^ source.terraform comment.line.number-sign.hcl punctuation.definition.comment.terraform
# ^^^^^^^^^^^^^^^^ source.terraform comment.line.number-sign.hcl
>
>variable "image_id" {
#^^^^^^^^ source.terraform meta.block.terraform entity.name.type.terraform
#        ^ source.terraform meta.block.terraform
#         ^ source.terraform meta.block.terraform entity.name.tag.terraform entity.name.tag.begin.terraform
#          ^^^^^^^^ source.terraform meta.block.terraform entity.name.tag.terraform
#                  ^ source.terraform meta.block.terraform entity.name.tag.terraform entity.name.tag.end.terraform
#                   ^ source.terraform meta.block.terraform
#                    ^ source.terraform meta.block.terraform punctuation.section.block.begin.terraform
>  type = string
#^^ source.terraform meta.block.terraform
#  ^^^^ source.terraform meta.block.terraform variable.declaration.terraform variable.other.readwrite.terraform
#      ^ source.terraform meta.block.terraform variable.declaration.terraform
#       ^ source.terraform meta.block.terraform variable.declaration.terraform keyword.operator.assignment.terraform
#        ^ source.terraform meta.block.terraform variable.declaration.terraform
#         ^^^^^^ source.terraform meta.block.terraform storage.type.terraform
>}
#^ source.terraform meta.block.terraform punctuation.section.block.end.terraform
>
>variable "availability_zone_names" {
#^^^^^^^^ source.terraform meta.block.terraform entity.name.type.terraform
#        ^ source.terraform meta.block.terraform
#         ^ source.terraform meta.block.terraform entity.name.tag.terraform entity.name.tag.begin.terraform
#          ^^^^^^^^^^^^^^^^^^^^^^^ source.terraform meta.block.terraform entity.name.tag.terraform
#                                 ^ source.terraform meta.block.terraform entity.name.tag.terraform entity.name.tag.end.terraform
#                                  ^ source.terraform meta.block.terraform
#                                   ^ source.terraform meta.block.terraform punctuation.section.block.begin.terraform
>  type    = list(string)
#^^ source.terraform meta.block.terraform
#  ^^^^ source.terraform meta.block.terraform variable.declaration.terraform variable.other.readwrite.terraform
#      ^^^^ source.terraform meta.block.terraform variable.declaration.terraform
#          ^ source.terraform meta.block.terraform variable.declaration.terraform keyword.operator.assignment.terraform
#           ^ source.terraform meta.block.terraform variable.declaration.terraform
#            ^^^^ source.terraform meta.block.terraform meta.function-call.terraform support.function.builtin.terraform
#                ^ source.terraform meta.block.terraform meta.function-call.terraform punctuation.section.parens.begin.terraform
#                 ^^^^^^ source.terraform meta.block.terraform meta.function-call.terraform storage.type.terraform
#                       ^ source.terraform meta.block.terraform meta.function-call.terraform punctuation.section.parens.end.terraform
>  default = ["us-west-1a"]
#^^ source.terraform meta.block.terraform
#  ^^^^^^^ source.terraform meta.block.terraform variable.declaration.terraform variable.other.readwrite.terraform
#         ^ source.terraform meta.block.terraform variable.declaration.terraform
#          ^ source.terraform meta.block.terraform variable.declaration.terraform keyword.operator.assignment.terraform
#           ^ source.terraform meta.block.terraform variable.declaration.terraform
#            ^ source.terraform meta.block.terraform punctuation.section.brackets.begin.terraform
#             ^ source.terraform meta.block.terraform string.quoted.double.terraform punctuation.definition.string.begin.terraform
#              ^^^^^^^^^^ source.terraform meta.block.terraform string.quoted.double.terraform
#                        ^ source.terraform meta.block.terraform string.quoted.double.terraform punctuation.definition.string.end.terraform
#                         ^ source.terraform meta.block.terraform punctuation.section.brackets.end.terraform
>}
#^ source.terraform meta.block.terraform punctuation.section.block.end.terraform
>
>variable "docker_ports" {
#^^^^^^^^ source.terraform meta.block.terraform entity.name.type.terraform
#        ^ source.terraform meta.block.terraform
#         ^ source.terraform meta.block.terraform entity.name.tag.terraform entity.name.tag.begin.terraform
#          ^^^^^^^^^^^^ source.terraform meta.block.terraform entity.name.tag.terraform
#                      ^ source.terraform meta.block.terraform entity.name.tag.terraform entity.name.tag.end.terraform
#                       ^ source.terraform meta.block.terraform
#                        ^ source.terraform meta.block.terraform punctuation.section.block.begin.terraform
>  type = list(object({
#^^ source.terraform meta.block.terraform
#  ^^^^ source.terraform meta.block.terraform variable.declaration.terraform variable.other.readwrite.terraform
#      ^ source.terraform meta.block.terraform variable.declaration.terraform
#       ^ source.terraform meta.block.terraform variable.declaration.terraform keyword.operator.assignment.terraform
#        ^ source.terraform meta.block.terraform variable.declaration.terraform
#         ^^^^ source.terraform meta.block.terraform meta.function-call.terraform support.function.builtin.terraform
#             ^ source.terraform meta.block.terraform meta.function-call.terraform punctuation.section.parens.begin.terraform
#              ^^^^^^ source.terraform meta.block.terraform meta.function-call.terraform meta.function-call.terraform support.function.builtin.terraform
#                    ^ source.terraform meta.block.terraform meta.function-call.terraform meta.function-call.terraform punctuation.section.parens.begin.terraform
#                     ^ source.terraform meta.block.terraform meta.function-call.terraform meta.function-call.terraform punctuation.section.braces.begin.terraform
>    internal = number
#^^^^ source.terraform meta.block.terraform meta.function-call.terraform meta.function-call.terraform
#    ^^^^^^^^ source.terraform meta.block.terraform meta.function-call.terraform meta.function-call.terraform variable.declaration.terraform variable.other.readwrite.terraform
#            ^ source.terraform meta.block.terraform meta.function-call.terraform meta.function-call.terraform variable.declaration.terraform
#             ^ source.terraform meta.block.terraform meta.function-call.terraform meta.function-call.terraform variable.declaration.terraform keyword.operator.assignment.terraform
#              ^ source.terraform meta.block.terraform meta.function-call.terraform meta.function-call.terraform variable.declaration.terraform
#               ^^^^^^ source.terraform meta.block.terraform meta.function-call.terraform meta.function-call.terraform storage.type.terraform
>    external = number
#^^^^ source.terraform meta.block.terraform meta.function-call.terraform meta.function-call.terraform
#    ^^^^^^^^ source.terraform meta.block.terraform meta.function-call.terraform meta.function-call.terraform variable.declaration.terraform variable.other.readwrite.terraform
#            ^ source.terraform meta.block.terraform meta.function-call.terraform meta.function-call.terraform variable.declaration.terraform
#             ^ source.terraform meta.block.terraform meta.function-call.terraform meta.function-call.terraform variable.declaration.terraform keyword.operator.assignment.terraform
#              ^ source.terraform meta.block.terraform meta.function-call.terraform meta.function-call.terraform variable.declaration.terraform
#               ^^^^^^ source.terraform meta.block.terraform meta.function-call.terraform meta.function-call.terraform storage.type.terraform
>    protocol = string
#^^^^ source.terraform meta.block.terraform meta.function-call.terraform meta.function-call.terraform
#    ^^^^^^^^ source.terraform meta.block.terraform meta.function-call.terraform meta.function-call.terraform variable.declaration.terraform variable.other.readwrite.terraform
#            ^ source.terraform meta.block.terraform meta.function-call.terraform meta.function-call.terraform variable.declaration.terraform
#             ^ source.terraform meta.block.terraform meta.function-call.terraform meta.function-call.terraform variable.declaration.terraform keyword.operator.assignment.terraform
#              ^ source.terraform meta.block.terraform meta.function-call.terraform meta.function-call.terraform variable.declaration.terraform
#               ^^^^^^ source.terraform meta.block.terraform meta.function-call.terraform meta.function-call.terraform storage.type.terraform
>  }))
#^^ source.terraform meta.block.terraform meta.function-call.terraform meta.function-call.terraform
#  ^ source.terraform meta.block.terraform meta.function-call.terraform meta.function-call.terraform punctuation.section.braces.end.terraform
#   ^ source.terraform meta.block.terraform meta.function-call.terraform meta.function-call.terraform punctuation.section.parens.end.terraform
#    ^ source.terraform meta.block.terraform meta.function-call.terraform punctuation.section.parens.end.terraform
>  default = [
#^^ source.terraform meta.block.terraform
#  ^^^^^^^ source.terraform meta.block.terraform variable.declaration.terraform variable.other.readwrite.terraform
#         ^ source.terraform meta.block.terraform variable.declaration.terraform
#          ^ source.terraform meta.block.terraform variable.declaration.terraform keyword.operator.assignment.terraform
#           ^ source.terraform meta.block.terraform variable.declaration.terraform
#            ^ source.terraform meta.block.terraform punctuation.section.brackets.begin.terraform
>    {
#^^^^ source.terraform meta.block.terraform
#    ^ source.terraform meta.block.terraform meta.braces.terraform punctuation.section.braces.begin.terraform
>      internal = 8300
#^^^^^^ source.terraform meta.block.terraform meta.braces.terraform
#      ^^^^^^^^ source.terraform meta.block.terraform meta.braces.terraform meta.mapping.key.terraform variable.other.readwrite.terraform
#              ^ source.terraform meta.block.terraform meta.braces.terraform
#               ^ source.terraform meta.block.terraform meta.braces.terraform keyword.operator.assignment.terraform
#                ^ source.terraform meta.block.terraform meta.braces.terraform
#                 ^^^^ source.terraform meta.block.terraform meta.braces.terraform constant.numeric.integer.terraform
>      external = 8300
#^^^^^^ source.terraform meta.block.terraform meta.braces.terraform
#      ^^^^^^^^ source.terraform meta.block.terraform meta.braces.terraform meta.mapping.key.terraform variable.other.readwrite.terraform
#              ^ source.terraform meta.block.terraform meta.braces.terraform
#               ^ source.terraform meta.block.terraform meta.braces.terraform keyword.operator.assignment.terraform
#                ^ source.terraform meta.block.terraform meta.braces.terraform
#                 ^^^^ source.terraform meta.block.terraform meta.braces.terraform constant.numeric.integer.terraform
>      protocol = "tcp"
#^^^^^^ source.terraform meta.block.terraform meta.braces.terraform
#      ^^^^^^^^ source.terraform meta.block.terraform meta.braces.terraform meta.mapping.key.terraform variable.other.readwrite.terraform
#              ^ source.terraform meta.block.terraform meta.braces.terraform
#               ^ source.terraform meta.block.terraform meta.braces.terraform keyword.operator.assignment.terraform
#                ^ source.terraform meta.block.terraform meta.braces.terraform
#                 ^ source.terraform meta.block.terraform meta.braces.terraform string.quoted.double.terraform punctuation.definition.string.begin.terraform
#                  ^^^ source.terraform meta.block.terraform meta.braces.terraform string.quoted.double.terraform
#                     ^ source.terraform meta.block.terraform meta.braces.terraform string.quoted.double.terraform punctuation.definition.string.end.terraform
>    }
#^^^^ source.terraform meta.block.terraform meta.braces.terraform
#    ^ source.terraform meta.block.terraform meta.braces.terraform punctuation.section.braces.end.terraform
>  ]
#^^ source.terraform meta.block.terraform
#  ^ source.terraform meta.block.terraform punctuation.section.brackets.end.terraform
>}
#^ source.terraform meta.block.terraform punctuation.section.block.end.terraform
>
>variable "image_id" {
#^^^^^^^^ source.terraform meta.block.terraform entity.name.type.terraform
#        ^ source.terraform meta.block.terraform
#         ^ source.terraform meta.block.terraform entity.name.tag.terraform entity.name.tag.begin.terraform
#          ^^^^^^^^ source.terraform meta.block.terraform entity.name.tag.terraform
#                  ^ source.terraform meta.block.terraform entity.name.tag.terraform entity.name.tag.end.terraform
#                   ^ source.terraform meta.block.terraform
#                    ^ source.terraform meta.block.terraform punctuation.section.block.begin.terraform
>  type        = string
#^^ source.terraform meta.block.terraform
#  ^^^^ source.terraform meta.block.terraform variable.declaration.terraform variable.other.readwrite.terraform
#      ^^^^^^^^ source.terraform meta.block.terraform variable.declaration.terraform
#              ^ source.terraform meta.block.terraform variable.declaration.terraform keyword.operator.assignment.terraform
#               ^ source.terraform meta.block.terraform variable.declaration.terraform
#                ^^^^^^ source.terraform meta.block.terraform storage.type.terraform
>  description = "The id of the machine image (AMI) to use for the server."
#^^ source.terraform meta.block.terraform
#  ^^^^^^^^^^^ source.terraform meta.block.terraform variable.declaration.terraform variable.other.readwrite.terraform
#             ^ source.terraform meta.block.terraform variable.declaration.terraform
#              ^ source.terraform meta.block.terraform variable.declaration.terraform keyword.operator.assignment.terraform
#               ^ source.terraform meta.block.terraform variable.declaration.terraform
#                ^ source.terraform meta.block.terraform string.quoted.double.terraform punctuation.definition.string.begin.terraform
#                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.terraform meta.block.terraform string.quoted.double.terraform
#                                                                         ^ source.terraform meta.block.terraform string.quoted.double.terraform punctuation.definition.string.end.terraform
>
>  validation {
#^^ source.terraform meta.block.terraform
#  ^^^^^^^^^^ source.terraform meta.block.terraform meta.block.terraform entity.name.label.terraform
#            ^ source.terraform meta.block.terraform meta.block.terraform
#             ^ source.terraform meta.block.terraform meta.block.terraform punctuation.section.block.begin.terraform
>    condition     = length(var.image_id) > 4 && substr(var.image_id, 0, 4) == "ami-"
#^^^^ source.terraform meta.block.terraform meta.block.terraform
#    ^^^^^^^^^ source.terraform meta.block.terraform meta.block.terraform variable.declaration.terraform variable.other.readwrite.terraform
#             ^^^^^ source.terraform meta.block.terraform meta.block.terraform variable.declaration.terraform
#                  ^ source.terraform meta.block.terraform meta.block.terraform variable.declaration.terraform keyword.operator.assignment.terraform
#                   ^ source.terraform meta.block.terraform meta.block.terraform variable.declaration.terraform
#                    ^^^^^^ source.terraform meta.block.terraform meta.block.terraform meta.function-call.terraform support.function.builtin.terraform
#                          ^ source.terraform meta.block.terraform meta.block.terraform meta.function-call.terraform punctuation.section.parens.begin.terraform
#                           ^^^ source.terraform meta.block.terraform meta.block.terraform meta.function-call.terraform support.constant.terraform
#                              ^ source.terraform meta.block.terraform meta.block.terraform meta.function-call.terraform keyword.operator.accessor.terraform
#                               ^^^^^^^^ source.terraform meta.block.terraform meta.block.terraform meta.function-call.terraform variable.other.member.terraform
#                                       ^ source.terraform meta.block.terraform meta.block.terraform meta.function-call.terraform punctuation.section.parens.end.terraform
#                                        ^ source.terraform meta.block.terraform meta.block.terraform
#                                         ^ source.terraform meta.block.terraform meta.block.terraform keyword.operator.terraform
#                                          ^ source.terraform meta.block.terraform meta.block.terraform
#                                           ^ source.terraform meta.block.terraform meta.block.terraform constant.numeric.integer.terraform
#                                            ^ source.terraform meta.block.terraform meta.block.terraform
#                                             ^^ source.terraform meta.block.terraform meta.block.terraform keyword.operator.logical.terraform
#                                               ^ source.terraform meta.block.terraform meta.block.terraform
#                                                ^^^^^^ source.terraform meta.block.terraform meta.block.terraform meta.function-call.terraform support.function.builtin.terraform
#                                                      ^ source.terraform meta.block.terraform meta.block.terraform meta.function-call.terraform punctuation.section.parens.begin.terraform
#                                                       ^^^ source.terraform meta.block.terraform meta.block.terraform meta.function-call.terraform support.constant.terraform
#                                                          ^ source.terraform meta.block.terraform meta.block.terraform meta.function-call.terraform keyword.operator.accessor.terraform
#                                                           ^^^^^^^^ source.terraform meta.block.terraform meta.block.terraform meta.function-call.terraform variable.other.member.terraform
#                                                                   ^ source.terraform meta.block.terraform meta.block.terraform meta.function-call.terraform punctuation.separator.terraform
#                                                                    ^ source.terraform meta.block.terraform meta.block.terraform meta.function-call.terraform
#                                                                     ^ source.terraform meta.block.terraform meta.block.terraform meta.function-call.terraform constant.numeric.integer.terraform
#                                                                      ^ source.terraform meta.block.terraform meta.block.terraform meta.function-call.terraform punctuation.separator.terraform
#                                                                       ^ source.terraform meta.block.terraform meta.block.terraform meta.function-call.terraform
#                                                                        ^ source.terraform meta.block.terraform meta.block.terraform meta.function-call.terraform constant.numeric.integer.terraform
#                                                                         ^ source.terraform meta.block.terraform meta.block.terraform meta.function-call.terraform punctuation.section.parens.end.terraform
#                                                                          ^ source.terraform meta.block.terraform meta.block.terraform
#                                                                           ^^ source.terraform meta.block.terraform meta.block.terraform keyword.operator.terraform
#                                                                             ^ source.terraform meta.block.terraform meta.block.terraform
#                                                                              ^ source.terraform meta.block.terraform meta.block.terraform string.quoted.double.terraform punctuation.definition.string.begin.terraform
#                                                                               ^^^^ source.terraform meta.block.terraform meta.block.terraform string.quoted.double.terraform
#                                                                                   ^ source.terraform meta.block.terraform meta.block.terraform string.quoted.double.terraform punctuation.definition.string.end.terraform
>    error_message = "The image_id value must be a valid AMI id, starting with \"ami-\"."
#^^^^ source.terraform meta.block.terraform meta.block.terraform
#    ^^^^^^^^^^^^^ source.terraform meta.block.terraform meta.block.terraform variable.declaration.terraform variable.other.readwrite.terraform
#                 ^ source.terraform meta.block.terraform meta.block.terraform variable.declaration.terraform
#                  ^ source.terraform meta.block.terraform meta.block.terraform variable.declaration.terraform keyword.operator.assignment.terraform
#                   ^ source.terraform meta.block.terraform meta.block.terraform variable.declaration.terraform
#                    ^ source.terraform meta.block.terraform meta.block.terraform string.quoted.double.terraform punctuation.definition.string.begin.terraform
#                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.terraform meta.block.terraform meta.block.terraform string.quoted.double.terraform
#                                                                              ^^ source.terraform meta.block.terraform meta.block.terraform string.quoted.double.terraform constant.character.escape.terraform
#                                                                                ^^^^ source.terraform meta.block.terraform meta.block.terraform string.quoted.double.terraform
#                                                                                    ^^ source.terraform meta.block.terraform meta.block.terraform string.quoted.double.terraform constant.character.escape.terraform
#                                                                                      ^ source.terraform meta.block.terraform meta.block.terraform string.quoted.double.terraform
#                                                                                       ^ source.terraform meta.block.terraform meta.block.terraform string.quoted.double.terraform punctuation.definition.string.end.terraform
>  }
#^^ source.terraform meta.block.terraform meta.block.terraform
#  ^ source.terraform meta.block.terraform meta.block.terraform punctuation.section.block.end.terraform
>}
#^ source.terraform meta.block.terraform punctuation.section.block.end.terraform
>